tests/dummy/app/helpers/boolean.js

```diff
@@ -0,0 +1,21 @@
+import Ember from 'ember';
+
+const { set, get } = Ember;
+
+class BoolState {
+  constructor(val) {
+    this.value = val;
+  }
+
+  get toggle() {
+    return () => {
+      set(this, 'value', !get(this, 'value'));
```

Comment:
I started out with that approach actually, and was surprised to find that it didn't work.

`Ember.set` seems to be the key in getting the template itself to realize that the underlying value has changed. Ember even gives this nifty error message in that case:

```Uncaught Error: Assertion Failed: You must use Ember.set() to set the `value` property (of [object Object]) to `true`.```